all travis-ci builds on Linux give false positives

If you look at a recent build that fails, like:

https://travis-ci.org/github/analogdevicesinc/libiio/builds/661315711?utm_source=github_status&utm_medium=notification

The Linux builds are all passing - which is wrong. Looking at the logs:

/docker_build_dir/dns_sd_avahi.c:384: warning: implicit declaration of function 'avahi_simple_poll_free'
/docker_build_dir/dns_sd_avahi.c:384: error: 'struct dns_sd_discovery_data' has no member named 'poll'
make[2]: *** [CMakeFiles/iio.dir/dns_sd_avahi.c.o] Error 1
make[1]: *** [CMakeFiles/iio.dir/all] Error 2
make: *** [all] Error 2

But the script continues and provides a 0 error code on exit.

(I think) since the exit code is dropped from the make, since the script library runs and rate checks things (which functions).

--- a/CI/travis/before_install_linux
+++ b/CI/travis/before_install_linux
@@ -19,7 +19,9 @@ handle_ubuntu_docker() {
 
 handle_default() {
 	sudo apt-get -qq update
-	sudo apt-get install -y cmake graphviz libaio-dev libavahi-client-dev libavahi-common-dev libusb-1.0-0-dev libxml2-dev rpm tar bzip2 gzip flex bison git
+	sudo apt-get install -y cmake graphviz libaio-dev libavahi-client-dev \
+		libavahi-common-dev libusb-1.0-0-dev libxml2-dev rpm tar \
+		bzip2 gzip flex bison git python-pip
 	if [ -n "${GH_DOC_TOKEN}" ] ; then
 		sudo apt-get install -y doxygen
 	fi

--- a/CI/travis/make_darwin
+++ b/CI/travis/make_darwin
@@ -9,5 +9,6 @@ ls
 
 cd $TRAVIS_BUILD_DIR/build_tar
 cmake -DOSX_PACKAGE=OFF -DENABLE_PACKAGING=ON -DPYTHON_BINDINGS=ON ..
-make && make package
+make
+make package
 ls

--- a/CI/travis/make_linux
+++ b/CI/travis/make_linux
@@ -24,7 +24,8 @@ handle_default() {
 	cmake -DENABLE_PACKAGING=ON -DDEB_DETECT_DEPENDENCIES=ON \
 		-DPYTHON_BINDINGS=ON ${DOC_FOUND} ..
 
-	make && make package
+	make
+	make package
 	if [ -n "${GH_DOC_TOKEN}" ] && \
 			[ -f "./generateDocumentationAndDeploy.sh" ] ; then
 		sh generateDocumentationAndDeploy.sh
@@ -36,7 +37,8 @@ handle_centos() {
 	mkdir -p build
 	cd build
 	cmake -DENABLE_PACKAGING=ON -DPYTHON_BINDINGS=ON ..
-	make && make package
+	make
+	make package
 	cd ..
 }
 

